tests/test.py

```diff
@@ -24,32 +23,61 @@
 CLIENT = docker.from_env()
 
 
-def version_arguments(*, image: str, volumes: dict, working_dir: str):
-    """Test the version arguments listed in the config"""
+def version_arguments(*, tool: str, environment: str):
```

Comment:
Testing functions like this one had to get much smarter so they would only test the parts that are relevant to a given tool/environment combination.  I did attempt to remove duplicate logic, but found that there were many edge cases or needs that were slightly different and made the value of making a centralized function much lower